pkg/repositories/joke.go

```diff
@@ -0,0 +1,124 @@
+package repositories
+
+import (
+	"chuck-jokes/models"
+	gormModels "chuck-jokes/pkg/database/gorm/models"
+	"log"
+	"time"
+
+	"gorm.io/gorm"
+)
+
+type JokeRepository interface {
+	JokeOfTheDay(time string) *models.Joke
+	GetJoke(jokeID uint) *models.Joke
+	GetStatistic(jokeID uint) (*models.Joke, uint)
+	GetJokes(page int, perPage int) *Pagination[models.Joke]
```

Comment:
FindAll(FindCollectionRequest) mysle ze taka sygnatura moze bardziej pasowac. FindCollectionRequest to struktura ktora moze miec page i perPage teraz ale w przyszlosci moze jakies dane dotyczace sorttowania itd.